### PR TITLE
Added part query function by unique id

### PR DESCRIPTION
**Pull request by peeter123**
# Description
Added part query function to get a part by unique id as defined in the Octopart V3 API: https://octopart.com/api/docs/v3/rest-api#endpoints-parts-get

# How was this tested
As given by the README of this repository.

If anything needs changing please let me know!

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -5876,3 +5876,1018 @@
...
+        "unit": null
+      },
+      "min_value": null,
+      "value": [
+        "Not Applicable"
+      ]
+    }
+  },
+  "uid": "eddc25bd5de8321b"
+}""")
```
Comment (antonf) [inline tests/fixtures.py]: Please add new line here.

Comment (antonf) [reply tests/fixtures.py]: Also, can you get rid of a bunch of Datasheets and PartOffers in added JSON to reduce line count a bit?

File: octopart/client.py [hunk 2]
```diff
@@ -198,6 +199,44 @@ def search(self,
...
+            uid (str): free-form keyword query
+
+        Kwargs:
+            includes, hide, show: Same as `match()`.
+
+        Returns:
+            dict. See `models.Part` for exact fields.
+        """
+
+        if not re.compile("^[a-f0-9]{16}$").match(uid):
```
Comment (antonf) [inline octopart/client.py]: Is there any reason not to use plain `re.match("^[a-f0-9]{16}$", uid)`? Also, to be consistent with code style, please use single quote strings unless there is good reason no to.

Comment (TadLeonard) [reply octopart/client.py]: For what it's worth, `re.match` will cache the compilation of the pattern you get it. For this reason, you can skip explicit `.compile` call for the most part.

**Comment by peeter123** [comment]
Thanks for your feedback, I've pushed some changes that should address these.

**Pull request merged**